LibGfx+Fuzz: Convert ImageDecoder::initialize to ErrorOr

BenWiederhake opened the pull request:
This prevents callers from accidentally discarding the result of initialize(), which was the root cause of this OSS Fuzz bug:

https://bugs.chromium.org/p/oss-fuzz/issues/detail?id=55896&q=label%3AProj-serenity&sort=summary

Note that `SpiceAgent.cpp::on_message_received() … case MessageType::Clipboard:` is still a mess, but now it might be slightly easier to clean up. :yak_bait:

nico commented on Tests/LibGfx/TestICCProfile.cpp:
```diff
@@ -60,7 +60,7 @@ TEST_CASE(webp_extended_lossy)
     auto file = MUST(Core::MappedFile::map(TEST_INPUT("extended-lossy.webp"sv)));
     auto webp = MUST(Gfx::WebPImageDecoderPlugin::create(file->bytes()));
-    EXPECT(webp->initialize());
+    EXPECT(!webp->initialize().is_error());
```
Use MUST() instead (also elsewhere)

BenWiederhake commented:
Changes since initial push: Convert all new `EXPECT\(!….is_error())` to `MUST(…)`. Thanks @nico!

BenWiederhake commented:
Changes since last push:
- Rebased, and resolved a conflict in ` Userland/Libraries/LibGfx/ImageFormats/WebPLoader.cpp ` due to the new `WebPImageDecoderPlugin::set_error` mechanism.

BenWiederhake commented:
Changes since last push: The new tests in `TestImageDecoder` call `initialize`, now I updated those callsites too.
